blueorgreenservice/src/main/java/org/springframework/demo/ColorController.java

```diff
@@ -8,40 +11,43 @@
  */
 @RestController
 public class ColorController {
-
+	private static final Logger LOGGER = LoggerFactory.getLogger(ColorController.class);
 	private ColorProperties colorProperties;
+	private Tracer tracer;
 
-	public ColorController(ColorProperties colorProperties) {
+	public ColorController(ColorProperties colorProperties, Tracer tracer) {
 		this.colorProperties = colorProperties;
+		this.tracer = tracer;
 	}
 
 	@GetMapping(path = {"/blueorgreen", "/"})
-	public Color color() throws InterruptedException {
+	public ColorResponse color() throws InterruptedException {
 		if (colorProperties.isSlow()) {
 			Thread.sleep(5000);
 		}
-		if (Color.BLUE.getId().equalsIgnoreCase(colorProperties.getColor())) {
-			return Color.BLUE;
-		} else if (Color.YELLOW.getId().equalsIgnoreCase(colorProperties.getColor())) {
-			return Color.YELLOW;
-		}
 
-		return Color.GREEN;
-	}
+		Color color = Color.valueOf(colorProperties.getColor());
+		tracer.currentSpan().tag("color", String.valueOf(color));
+		LOGGER.info("Picked color: {}", color);
```

Comment:
This is important so that you will see the traceId/spanId in the logs and correlate/search in wavefront, also, the tag of the span will tell you which service was involved (the name of the services are the same).